packages/rspack/src/config/zod/_rewrite.ts

```diff
@@ -19,7 +19,8 @@ type Rewritten = {
 	resolve?: Resolve;
 	module?: ModuleOptions;
 	plugins?: Plugins;
-	builtins?: Builtins;
+	builtins?: Omit<Builtins, keyof NonNullable<Config["builtins"]>> &
```

Comment:
What's the meaning of `NonNullable` for an object? Looks no difference with the parameter itself to me.

Reply:
The type of `Config["builtins"]` is `{ xxx: ...} | undefined`. `keyof { xxx: ...} | undefined` would resolve to `never`. `NonNullable<Config["builtins"]>` would filter out `undefined`.

Reply:
Make sense